src/v/storage/tests/utils/random_batch.cc

```diff
@@ -188,7 +188,7 @@ make_random_batches(record_batch_spec spec) {
     model::offset o = spec.offset;
     int32_t base_sequence = spec.base_sequence;
     for (int i = 0; i < spec.count; i++) {
-        auto num_records = get_int(2, 30);
+        auto num_records = spec.records ? *spec.records : get_int(2, 30);
```

Comment:
I do not know, but is it better use `has_value()`, `value()` ?

Reply:
```
auto num_records = spec.records.value_or(get_int(2, 30));
```
But only if `get_int(2, 30)` is cheap. Something, something, monads, probably.

Reply:
http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p0798r6.html